### PR TITLE
Restore login screens’ bg tap gesture recognizer

### LINKED ISSUE #8651: The keyboard is not dismissible during the LogIn flow
### Expected behavior
I should be able to dismiss the keyboard during the LogIn process.

### Actual behavior
Tapping outside the email text view does not dismiss the keyboard.

### Steps to reproduce the behavior
Check out `develop`, trigger Log In, start entering text and then tap outside the text view, nothing will happen.

##### Tested on [device], iOS [version], WPiOS [version]
Simulator, 11, 9.5

cc: @nheagy @jleandroperez

### PR DESCRIPTION
**Pull request by nheagy**
Fixes #8651 

Tapping the background in a login screen will once again dismiss the keyboard

To test:
- start logging in
- tap a text field to bring up the keyboard
- tap anywhere empty to make the keyboard go away

Note:
I reorganized my growing cache of extensions for `@objc`-less, block-based events to clarify the license coverage, and give our `NaturalContentHorizontalAlignment` a more natural home.

### CODE REVIEW

File: WordPressUI/WordPressUI/Extensions/Events without selectors/UIBarButtonItem+BlockEvents.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Adding UIBarButtonItem
```
Comment (elibud) [inline WordPressUI/WordPressUI/Extensions/Events without selectors/UIBarButtonItem+BlockEvents.swift]: We probably want to include the reference to the GitHub repo as well as the license here as well.

Comment (nheagy) [reply WordPressUI/WordPressUI/Extensions/Events without selectors/UIBarButtonItem+BlockEvents.swift]: I wrote that code, I assume it's under our license.

Comment (elibud) [reply WordPressUI/WordPressUI/Extensions/Events without selectors/UIBarButtonItem+BlockEvents.swift]: haha under Nate's license, for some reason I thought it all came from the same repo, sorry! 🤦‍♀️

**Comment by nheagy** [comment]
Great feedback Eli!

I went with `BlockEvents` instead of `EventsWithoutSelectors`, I hope that's ok. The license is already in the right spot; Ian only wrote the `UIControl` case, the two others are extensions I wrote based on his code. I think we're on the clear for the license.

Ready for you again @elibud!

**Comment by jleandroperez** [comment]
@nheagy thank you so much for pushing this into the `WordPressUI` target!!!

❤️

**Pull request merged**